### PR TITLE
Handle docs, tools

### PR DESCRIPTION
**Pull request by djmitche**
OK, there's a *lot* here, sorry.  It's split up into commits, but even some of those are quite large.

The tl;dr here is

* lots of factoring of the build process
* switch to building repos, instead of services, and include all the docs-only repos
* build docs metadata for services using `docker run write-docs` to write it to a directory
* marshall all the docs into the tools site
* build the tools site
* deploy the tools site (sort of)

Things I'd like @imbstack and @jonasfj to be aware of, as folks likely to hack on this repo in the future:
* how dependencies work for build tasks (named values, usually with ${name} in the name)
* how build tasks "skip" (generally by looking for existing output)
* "split" process of building services: first build the slug (buildpack compile), then build an image from that; the idea is that we can later stick an extra step in here to include schemas and references at some fixed location in the FS, thus making updated metadata available to clients.
* I haven't paid much attention to deployment here..

It's Friday so I don't need to land this immediately, but I'd like to land it and move on without waiting for a detailed review.

### CODE REVIEW

**Comment by djmitche** [comment]
I added some more -- a few command-line options, some better cloning, and a good bit of documentation.

At this point, for me the entire build process skips in just a few seconds if there are no changes, but any change to the source repositories will result in a rebuild of the relevant bits (always including tools, which takes a *long* time to build).

Note that changes to the build process itself do not trigger a rebuild.  I may add "clobber" support to help with that later.

File: docs/build-implementation.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+title: Build Implementatoin
```
Comment (imbstack) [inline docs/build-implementation.md]: More support for my "words that end in -ion are the funniest typos" theory.

File: docs/cluster-spec.md [hunk 2]
```diff
@@ -10,13 +10,36 @@ The contents are the same, but the `.yml` format is intended to be human-editabl
...
 
-* `services` -- an array of objects describing the services that must be built to generate a release; each has
-  * `name` -- the service name (without a `taskcluster-` prefix)
-  * `source` -- the source for the service, in `gitRepo#ref` format
-  * `exactSource` -- (generated by build process) similar to `source`, but with a commit sha1 as the reference
-  * `dockerImage` -- (generated by build process) the docker image containing this service
+Each element in this array has:
+
+* `name` -- the short name (for services, without a `taskcluster-` prefix; for libraries, the name used with `yarn add`)
+* `kind` -- the kind of repository, in othre words how it should be built
```
Comment (imbstack) [inline docs/cluster-spec.md]: othre
